src/System.Security.Claims/src/System/Security/Claims/ClaimsIdentity.cs

```diff
@@ -173,7 +173,7 @@ public ClaimsIdentity(IIdentity identity, IEnumerable<Claim> claims, string auth
         {
             ClaimsIdentity claimsIdentity = identity as ClaimsIdentity;
 
-            _authenticationType = !string.IsNullOrWhiteSpace(authenticationType) ? authenticationType : (identity != null ? identity.AuthenticationType : null);
+            _authenticationType = (identity != null && string.IsNullOrEmpty(authenticationType)) ? identity.AuthenticationType : authenticationType;
```

Comment:
I'm guessing that we can't allow WindowsIdentity's UnauthorizedAccessException => null due to layering?

I guess if we ever get a complaint about it we can make this more complicated, like always eating a UAE.